packages/core/src/browser/branding.ts

```diff
@@ -0,0 +1,103 @@
+/********************************************************************************
+ * Copyright (C) 2019 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Theme } from './theming';
+import { SharedStyle } from './shared-style';
+
+export interface BrandingTheme {
+
+    /**
+     * The main icon for the application.
+     */
+    mainIcon: (theme: Theme) => string,
+
+    /**
+     * The menu icon for the application.
+     */
+    menuIcon: (theme: Theme) => string,
+
+    /**
+     * The main icon size.
+     */
+    mainSize: string;
+}
+
+export class BrandingService {
+
+    protected style: SharedStyle;
+
+    protected readonly GETTING_STARTED_SELECTOR = '#getting-started-logo';
+    protected readonly MAIN_SELECTOR = '#theia-main-content-panel';
+    protected readonly MENU_SELECTOR = '.theia-icon';
+
+    protected brandingTheme: BrandingTheme;
```

Comment:
Can this be undefined?
Looks like this would crash, if I don't have a branding plug-in installed.

Reply:
Maybe we move keep the default values in core, but just allow an easy way to adjust the icons, through BrandingService. white labeled applications will likely want to adjust other things as well. Using `SharedStyle` in those cases seem appropriate as it offers all posibilities.

Reply:
If I move the values back into `core`, I assume there is no longer a need for 
the `default-branding` extension?

Core will now have the functionality to update the branding, while also providing the default branding icons?